cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -278,4 +331,66 @@ private String getRequestingUser() throws IllegalArgumentException {
     LOG.trace("Got requesting principal {}", principal);
     return principal.getName();
   }
+
+  @VisibleForTesting
+  public Map<AuthorizationPrivilege, Boolean> cacheAsMap() {
```

Comment:
The cache scheme for the container _is_ internal. If we push this class to the SPI or any place external then either it would be really difficult to change that or we will have to create a new class for that key. We are already talking about changing the cache scheme based on the type of container. So, this class will very likely change, that will become very difficult to do if it was part of the SPI. Sentry also has client-side caching on their roadmap, if and when that becomes available we might not even need the cache here and then we will end up with a class polluting the SPI that no one is using.